uptime.go

```diff
@@ -0,0 +1,324 @@
+package godo
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"path"
+)
+
+const uptimeChecksBasePath = "/v2/uptime/checks"
+
+// UptimeChecksService is an interface for creating and managing Uptime checks with the DigitalOcean API.
+// See: https://docs.digitalocean.com/reference/api/api-reference/#tag/Projects
+type UptimeChecksService interface {
+	List(context.Context, *ListOptions) ([]UptimeCheck, *Response, error)
+	Get(context.Context, string) (*UptimeCheck, *Response, error)
+	GetState(context.Context, string) (*UptimeCheck, *Response, error)
+	Create(context.Context, *CreateUptimeCheckRequest) (*UptimeCheck, *Response, error)
+	Update(context.Context, string, *UpdateUptimeCheckRequest) (*UptimeCheck, *Response, error)
+	Delete(context.Context, string) (*Response, error)
+	GetAlert(context.Context, string, string) (*Alert, *Response, error)
+	ListAlerts(context.Context, *ListOptions, string) ([]Alert, *Response, error)
+	CreateAlert(context.Context, *CreateAlertRequest, string) (*Alert, *Response, error)
+	UpdateAlert(context.Context, string, string, *UpdateAlertRequest) (*Alert, *Response, error)
+	DeleteAlert(context.Context, string, string) (*Response, error)
+}
+
+// UptimeChecksServiceOp handles communication with Uptime Check methods of the DigitalOcean API.
+type UptimeChecksServiceOp struct {
+	client *Client
+}
+
+// UptimeCheck represents a DigitalOcean UptimeCheck configuration.
+type UptimeCheck struct {
+	ID      string   `json:"id"`
+	Name    string   `json:"name"`
+	Type    string   `json:"type"`
+	Target  string   `json:"target"`
+	Regions []string `json:"regions"`
+	Enabled bool     `json:"enabled"`
+}
+
+// Alert represents a DigitalOcean Alert configuration.
+type Alert struct {
```

Comment:
Can we namespace these various Alert structs? e.g. `UptimeAlert`  We already have some service specific alerts like for Apps: 

https://github.com/digitalocean/godo/blob/7dec38659ec7e71f63797c8ee75f0f11f3a0259f/apps.gen.go#L10

And there is also the more generic `AlertPolicy` for the monitoring service:

https://github.com/digitalocean/godo/blob/16f4ba922214ad74129b1a4136d88dc94edaa402/monitoring.go#L81

I'd expect we'll have more types like these in the future, so namespacing them should help keep it from getting too confusing.

Reply:
good call 👍